Using the state hook traduction (#32)
reactjs/fr.reactjs.org: content/docs/hooks-state.md

---
id: hooks-state
title: Utiliser le Hook State
permalink: docs/hooks-state.html
next: hooks-effect.html
prev: hooks-overview.html
---

> Les *Hooks* sont la grande nouveauté de React 16.8. Ils permettent d'utiliser l'état et d'autres fonctionnalités React sans avoir à écrire une classe.

Author: Effectivement, je vois ce que tu veux dire, j'avais choisi cette traduction vis-à-vis de l'enthousiasme de beaucoup pour les Hooks :) et je trouvais que ça rendait mieux que `Les *Hooks* sont la nouveauté de React 16.8` ! Mais je peux changer ça sans soucis si c'est gênant !

> La [page précédente](/docs/hooks-intro.html) introduit les Hooks avec cet exemple :

Reviewer: ```suggestion
La [page précédente](/docs/hooks-intro.html) présentait les Hooks avec cet exemple :
```

```js{4-5}
import React, { useState } from 'react';

function Example() {
  // Déclare une nouvelle variable d'état, que l'on va appeler « count »
  const [count, setCount] = useState(0);

  return (
    <div>
      <p>Vous avez cliqué {count} fois</p>
      <button onClick={() => setCount(count + 1)}>

>         Cliquez moi

Author: Je m'étais faite la même réflexion ;) mais si besoin, je peux changer !

>         Cliquez moi

Reviewer: ```suggestion
        Cliquez ici
```

En fait on a normalisé partout sur la forme clean…

      </button>
    </div>
  );
}
```

> Comparons ce code avec un exemple équivalent utilisant une classe pour commencer à comprendre comment fonctionnent les Hooks.

Reviewer: ```suggestion
Pour commencer à comprendre comment fonctionnent les Hooks, comparons ce code avec un exemple équivalent à base de classe.
```

## Exemple équivalent avec une classe {#equivalent-class-example}

> Si vous avez déjà utilisé les classes en React, ce code devrait vous être familier :

Reviewer: ```suggestion
Si vous avez déjà utilisé les classes en React, ce code devrait vous parler :
```

  render() {
    return (
      <div>
        <p>Vous avez cliqué {this.state.count} fois</p>
        <button onClick={() => this.setState({ count: this.state.count + 1 })}>

>           Cliquez moi

Reviewer: ```suggestion
          Cliquez ici
```

        </button>
      </div>
    );
  }
}
```

> L'état vaut d'abord `{ count: 0 }`, et nous incrémenterons `state.count` en appelant `this.setState()` lorsque l'utilisateur cliquera sur le bouton. Dans la suite de cette page, nous réutiliserons des extraits de cette classe.

Reviewer: ```suggestion
L'état démarre à `{ count: 0 }`, et nous incrémentons `state.count` en appelant `this.setState()` lorsque l'utilisateur clique sur le bouton. Nous utiliserons des extraits de cette classe dans tout le reste de cette page.
```

> >Remarque :

Reviewer: ```suggestion
>Remarque
```

On normalise sans les deux-points un peu partout

>

> >Vous vous demandez peut-être pourquoi nous utilisons un compteur plutôt qu'un exemple plus réaliste. Ça nous permet tout simplement de nous concentrer sur l'API pendant que nous faisons nos premiers pas avec les Hooks.

Reviewer: GG 💯

## Hooks et fonctions composants {#hooks-and-function-components}

> Pour rappel, les fonctions composants dans React peuvent ressembler à ceci :

Reviewer: ```suggestion
Pour rappel, les fonctions composants en React ressemblent à ceci :
```

```js
const Example = (props) => {
  // Vous pouvez utiliser des Hooks ici !
  return <div />;
}
```

> ou cela :

Reviewer: ```suggestion
ou à ça :
```

```js
function Example(props) {
  // Vous pouvez utiliser des Hooks ici !
  return <div />;
}
```

> Vous les connaissiez peut-être sous le nom de « composants sans état ». Comme nous allons maintenant introduire la possibilité d'utiliser l'état React avec ces composants, nous préférerons utiliser le terme « fonctions composants ».

Reviewer: ```suggestion
Vous les connaissiez peut-être sous le nom de « composants sans état ». Comme nous avons maintenant la possibilité d'utiliser l'état local React dans ces composants, nous préférerons le terme « fonctions composants ».
```

> Les Hooks **ne fonctionnent pas** dans les classes. Mais vous pouvez les utiliser pour éviter d'écrire des classes.

Reviewer: Parfait 😎

}
```

> **Qu'est-ce qu'un Hook ?** Un Hook est une fonction qui permet de se « brancher » (*”hook into“ en anglais*, NdT) à des fonctionnalités React. Par exemple, `useState` est un Hook qui permet d'ajouter l'état local React à des fonctions composants. Nous en apprendrons plus sur les Hooks par la suite.

Reviewer: ```suggestion
**Qu'est-ce qu'un Hook ?** Un Hook est une fonction qui permet de « se brancher » sur des fonctionnalités React. Par exemple, `useState` est un Hook qui permet d'ajouter l'état local React à des fonctions composants. Nous en apprendrons plus sur les Hooks par la suite.
```

NdT inutile. Bon choix, j'avais fait le même en revue de PR ailleurs sur les hooks 😉 En-dehors de ça, rien à dire 👏

> **Quand utiliser un Hook ?** Auparavant, si vous écriviez une fonction composant et que vous réalisiez que vous aviez besoin d'un state à l'intérieur, il fallait le convertir en classe. Maintenant, vous pouvez utiliser un Hook à l'intérieur de votre fonction composant. Et c'est justement ce que nous allons faire !

Reviewer: ```suggestion
**Quand utiliser un Hook ?** Auparavant, si vous écriviez une fonction composant et que vous réalisiez que vous aviez besoin d'un état local à l'intérieur, vous deviez la convertir en classe. Désormais vous pouvez utiliser un Hook à l'intérieur de votre fonction composant. Et c'est justement ce que nous allons faire !
```

Micro-tweaks, mais nickel 👍

> >Remarque :

Reviewer: ```suggestion
>Remarque
```

>
>Des règles spécifiques existent pour savoir quand utiliser ou ne pas utiliser les Hooks dans un composant. Nous les découvrirons dans les [Règles des Hooks](/docs/hooks-rules.html).

## Déclarer une variable d'état {#declaring-a-state-variable}

> Dans une classe, on initialise le state `count` à `0` en initialisant `this.state` à `{ count : 0 }` dans le constructeur :

Reviewer: ```suggestion
Dans une classe, on initialise l’état local `count` à `0` en définissant `this.state` à `{ count: 0 }` dans le constructeur :
```

  }
```

> Dans une fonction composant, nous ne pouvons pas assigner ou lire `this.state` puisqu'il n'y a pas de `this`. Au lieu de ça, nous appellerons directement le Hook `useState` dans notre composant :

Reviewer: ```suggestion
Dans une fonction composant, nous ne pouvons pas écrire ou lire `this.state` puisqu'il n'y a pas de `this`. Au lieu de ça, nous appelons directement le Hook `useState` dans notre composant :
```

```js{4,5}
import React, { useState } from 'react';

function Example() {
  // Déclare une nouvelle variable d'état, que nous appellerons « count »
  const [count, setCount] = useState(0);
```

> **Appeler `useState`, qu'est-ce que ça fait ?** Cela déclare une « variable d'état ». Notre variable est appelée `count` mais nous aurions pu l'appeler n'importe comment, par exemple `banane`. C'est un moyen de « préserver » des valeurs entre différents appels de fonctions. `useState` est un moyen d'utiliser exactement les mêmes possibilités qu'offrent `this.state` dans une classe. Normalement, les variables « disparaissent » quand la fonction s'achève mais les variables d'état sont préservées par React.

Reviewer: ```suggestion
**Appeler `useState`, qu'est-ce que ça fait ?** Cela déclare une « variable d'état ». Notre variable est appelée `count` mais nous aurions pu l'appeler n'importe comment, par exemple `banane`. C'est un moyen de « préserver » des valeurs entre différents appels de fonctions. `useState` est une nouvelle façon d'utiliser exactement les mêmes possibilités qu'offre `this.state` dans une classe. Normalement, les variables « disparaissent » quand la fonction s'achève mais les variables d'état sont préservées par React.
```

Franchement, c'est impec'… juste une micro-retouche d'accord.

> **Qu'est-ce que l'on passe à `useState` comme argument ?** Le seul argument à passer au Hook `useState()` est l'état initial. Contrairement à son usage dans les classes, l'état n'est pas obligatoirement un objet. Il peut être un nombre ou une chaîne de caractères si c'est tout ce dont on a besoin. Dans notre exemple, nous voulons simplement un nombre traduisant le nombre de fois qu'un utilisateur a cliqué sur un bouton, nous passerons donc `0` comme état initial pour notre variable. (Si nous voulions stocker deux valeurs différentes dans l'état, nous aurions appelé `useState()` deux fois.)

Reviewer: ```suggestion
**Qu'est-ce qu'on passe à `useState` comme argument ?** Le seul argument à passer au Hook `useState()` est l'état initial. Contrairement à ce qui se passe dans les classes, l'état local n'est pas obligatoirement un objet. Il peut s’agir d’un nombre ou d’une chaîne de caractères si ça nous suffit. Dans notre exemple, nous voulons simplement un nombre traduisant le nombre de fois qu'un utilisateur a cliqué sur le bouton, nous passerons donc `0` comme état initial pour notre variable. (Si nous voulions stocker deux valeurs différentes dans l'état, nous appellerions `useState()` deux fois.)
```

> **Qu'est-ce que renvoie `useState` ?** Elle renvoie une paire de valeurs : l'état actuel et une fonction pour le modifier. C'est pourquoi nous écrivons `const [count, setCount] = useState()`. C'est semblable au `this.state.count` et `this.setState` dans une classe mais ici nous les récupérons en même temps. Si vous n'êtes pas familier avec la syntaxe que nous avons utilisé, nous y reviendrons [plus tard sur cette page](/docs/hooks-state.html#tip-what-do-square-brackets-mean).

Reviewer: ```suggestion
**Que renvoie `useState` ?** Elle renvoie une paire de valeurs : l'état actuel et une fonction pour le modifier. C'est pourquoi nous écrivons `const [count, setCount] = useState()`. C'est semblable à `this.state.count` et `this.setState` dans une classe, mais ici nous les récupérons en même temps. Si vous n'êtes pas à l’aise avec la syntaxe que nous avons employée, nous y reviendrons [en bas de cette page](/docs/hooks-state.html#tip-what-do-square-brackets-mean).
```

Maintenant que nous savons ce que fait le Hook `useState`, notre exemple devrait commencer à être plus clair :

```js{4,5}
import React, { useState } from 'react';

function Example() {
  // Déclare une nouvelle variable d'état, que nous appellerons « count »
  const [count, setCount] = useState(0);
```

> Nous déclarons une variable appelée `count`, et l'initialisons à `0`. React se rappellera sa valeur entre deux re-renders et fournira la plus récente à notre fonction. Si nous voulons modifier la valeur de `count`, nous pouvons utiliser `setCount`.

Reviewer: ```suggestion
Nous déclarons une variable d‘état appelée `count`, et l'initialisons à `0`. React se rappellera sa valeur entre deux affichages et fournira la plus récente à notre fonction. Si nous voulons modifier la valeur de `count`, nous pouvons appeler `setCount`.
```

> >Remarque :

Reviewer: ```suggestion
>Remarque
```

>
>Vous vous demandez peut-être pourquoi `useState` n'est pas plutôt appelée `createState` ?
>

> >En fait, « create » ne serait pas correct puisque ce n'est que lorsque notre composant est rendu pour la première fois que l'état est créé. Dans les rendus suivants, `useState` nous renvoie l'état actuel. Autrement, ce ne serait pas un état du tout ! Il y a aussi une autre raison pour laquelle les noms des Hooks commencent *toujours* par `use`. Nous découvrirons quelle est cette raison plus tard dans les [Règles des Hooks](/docs/hooks-rules.html).

Reviewer: ```suggestion
>En fait, « create » ne serait pas tout à fait correct puisque l'état n’est créé qu’au premier affichage de notre composant. Les fois suivantes, `useState` nous renvoie l'état actuel. Autrement, ce ne serait pas un état du tout ! Il y a aussi une raison pour laquelle les noms des Hooks commencent *toujours* par `use`. Nous découvrirons laquelle plus tard dans les [Règles des Hooks](/docs/hooks-rules.html).
```

## Lire l'état {#reading-state}

Quand nous voulons afficher la valeur actuelle de `count` dans une classe, nous récupérons la valeur de `this.state.count` :

```js
  <p>Vous avez cliqué {this.state.count} fois</p>
```

Dans une fonction, nous pouvons directement utiliser `count` :

```js
  <p>Vous avez cliqué {count} fois</p>
```

## Mettre à jour l'état {#updating-state}

Dans une classe, nous devons appeler `this.setState()` pour mettre à jour l'état `count` :

```js{1}
  <button onClick={() => this.setState({ count: this.state.count + 1 })}>

>     Cliquez moi

Reviewer: ```suggestion
    Cliquez ici
```

  </button>
```

> Dans une fonction, nous récupérons directement `setCount` et `count` comme variables, nous n'avons donc pas besoin d'utiliser `this` :

Reviewer: ```suggestion
Dans une fonction, nous récupérons directement `setCount` et `count` comme variables, nous n'avons donc pas besoin de `this` :
```

```js{1}
  <button onClick={() => setCount(count + 1)}>

>     Cliquez moi

Reviewer: ```suggestion
    Cliquez ici
```

  </button>
```

> ## Résumé {#recap}

Reviewer: ```suggestion
## En résumé {#recap}
```

> Il est maintenant temps de **résumer ce que nous avons appris ligne par ligne** et vérifier que nous avons bien compris.

Reviewer 2: ```suggestion
Il est maintenant temps de **récapituler ce que nous avons appris ligne par ligne** et vérifier que nous avons bien compris.
```

Reviewer 2: Juste pour éviter la répétition phonétique avec le titre

 5:
 6:    return (
 7:      <div>
 8:        <p>Vous avez cliqué {count} fois</p>
 9:        <button onClick={() => setCount(count + 1)}>

> 10:         Cliquez moi

Reviewer 2: et 10 de der. 🤣

Reviewer: Je l'espérais tellement ce 3e comment' 😉

> 10:         Cliquez moi

Reviewer: ```suggestion
10:         Cliquez ici
```

11:        </button>
12:      </div>
13:    );
14:  }
```

> * **Ligne 1 :** Nous importons le Hook `useState` de React. Il nous permet d'utiliser un état local dans une fonction composant. 

Reviewer: ```suggestion
* **Ligne 1 :** nous importons le Hook `useState` depuis React. Il nous permet d'utiliser un état local dans une fonction composant. 
```

On ne met pas de majuscules après un deux-points en français.

> * **Ligne 4 :** Dans le composant `Example`, nous déclarons une nouvelle variable d'état en appelant le Hook `useState`. Il renvoie une paire de valeurs que nous pouvons nommer à notre guise. Ici, nous appelons notre variable `count` puisqu'elle contient le nombre de clics sur le bouton. Nous l'initialisons à zéro en passant `0` comme seul argument à `useState`. Le second élément renvoyé est une fonction. Il nous permet de modifier la variable `count`, nous l'appellerons donc `setCount`.

Reviewer: ```suggestion
* **Ligne 4 :** dans le composant `Example`, nous déclarons une nouvelle variable d'état en appelant le Hook `useState`. Il renvoie une paire de valeurs que nous pouvons nommer à notre guise. Ici, nous appelons notre variable `count` puisqu'elle contient le nombre de clics sur le bouton. Nous l'initialisons à zéro en passant `0` comme seul argument à `useState`. Le second élément renvoyé est une fonction. Elle nous permet de modifier la variable `count`, nous l'appellerons donc `setCount`.
```

Au top !

> * **Ligne 9 :** Quand l'utilisateur clique, nous appelons `setCount` avec une nouvelle valeur. React refera le rendu du composant `Example` et lui passera la nouvelle valeur de `count`.

Reviewer: ```suggestion
* **Ligne 9 :** quand l'utilisateur clique, nous appelons `setCount` avec une nouvelle valeur. React rafraîchira le composant `Example` et lui passera la nouvelle valeur de `count`.
```

> C'est peut-être beaucoup à digérer d'un coup. Ne vous pressez pas ! Si vous êtes confus, jetez un nouveau coup d'oeil au code ci-dessus et relisez le en entier. Promis, une fois que vous essaierez « d'oublier » la manière dont fonctionne l'état dans les classes, et que vous regarderez ce code avec un regard neuf, ça sera plus clair.

Reviewer 2: ```suggestion
C'est peut-être beaucoup à digérer d'un coup. Ne vous pressez pas ! Si vous vous sentez un peu perdu, jetez un nouveau coup d’œil au code ci-dessus et essayez de le relire en entier. Promis, une fois que vous essaierez « d'oublier » la manière dont fonctionne l'état dans les classes, et que vous regarderez ce code avec un regard neuf, ça sera plus clair.
```

Reviewer 2: J'aime beaucoup l'utilisation de "confus" dans ce contexte, c'est très élégant, cependant c'est peut être un peut trop littéraire ici.

> C'est peut-être beaucoup à digérer d'un coup. Ne vous pressez pas ! Si vous vous sentez un peu perdu·e·, jetez un nouveau coup d'œil au code ci-dessus et essayez de le relire en entier. Promis, une fois que vous essaierez « d'oublier » la manière dont fonctionne l'état dans les classes, et que vous regarderez ce code avec un regard neuf, ça sera plus clair.

Reviewer: ```suggestion
Ça fait peut-être beaucoup à digérer d'un coup. Ne vous pressez pas ! Si vous vous sentez un peu perdu·e, jetez un nouveau coup d'œil au code ci-dessus et essayez de le relire du début à la fin. Promis, une fois que vous essaierez « d'oublier » la manière dont fonctionne l'état local dans les classes, et que vous regarderez ce code avec un regard neuf, ça sera plus clair.
```

Un plaisir

> ### Astuce : Que signifient les crochets ? {#tip-what-do-square-brackets-mean}

Reviewer: ```suggestion
### Astuce : que signifient les crochets ? {#tip-what-do-square-brackets-mean}
```

Vous avez peut-être remarqué les crochets que nous utilisons lorsque nous déclarons une variable d'état :

```js
  const [count, setCount] = useState(0);
```

> Les noms utilisés dans la partie gauche ne font pas partie de l'API React. Vous pouvez nommer vos variables d'état comme bon vous chante :

Reviewer: ```suggestion
Les noms utilisés dans la partie gauche ne font pas partie de l'API React. Vous pouvez nommer vos variables d'état comme ça vous chante :
```

🤣 « comme bon vous semble » ou « comme ça vous chante ». J'adore les mélanges d'expressions par inadvertance, genre « mettre de l'eau dans son gaz » 🤣

```js
  const [fruit, setFruit] = useState('banane');
```

> Cette syntaxe Javascript est appelée [« déstructuration positionnelle »](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring). Cela signifie que nous sommes en train de créer deux nouvelles variables `fruit` et `setFruit`, où la première valeur renvoyée par `useState` est affectée à `fruit`, et la deuxième à `setFruit`. C'est équivalent au code ci-dessous :

Reviewer 2: ```suggestion
Cette syntaxe Javascript est appelée [« déstructuration positionnelle »](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Op%C3%A9rateurs/Affecter_par_d%C3%A9composition#D%C3%A9composition_d'un_tableau). Cela signifie que nous sommes en train de créer deux nouvelles variables `fruit` et `setFruit`, où la première valeur renvoyée par `useState` est affectée à `fruit`, et la deuxième à `setFruit`. C'est équivalent au code ci-dessous :
```

Reviewer 2: Je propose le liens vers MDN en français dans le mesure ou même s'il y a à redire sur certain choix de traduction, la qualité de la doc en Français n'est plus à démontrer (au fait @tdd, n'hésite pas à corriger MDN si tu veux améliorer le vocabulaire utilisé la-bas 😜)

> Cette syntaxe Javascript est appelée [« déstructuration positionnelle »](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Op%C3%A9rateurs/Affecter_par_d%C3%A9composition#D%C3%A9composition_d'un_tableau). Cela signifie que nous sommes en train de créer deux nouvelles variables `fruit` et `setFruit`, où la première valeur renvoyée par `useState` est affectée à `fruit`, et la deuxième à `setFruit`. C'est équivalent au code ci-dessous :

Reviewer: ```suggestion
Cette syntaxe Javascript est appelée [« déstructuration positionnelle »](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Op%C3%A9rateurs/Affecter_par_d%C3%A9composition#D%C3%A9composition_d'un_tableau). Ça signifie que nous créons deux nouvelles variables `fruit` et `setFruit`, avec `fruit` qui reçoit la première valeur renvoyée par `useState`, et `setFruit` qui reçoit la deuxième. C'est équivalent au code ci-dessous :
```

```js
  var fruitStateVariable = useState('banana'); // Renvoie une paire
  var fruit = fruitStateVariable[0]; // Premier élément dans une paire
  var setFruit = fruitStateVariable[1]; // Deuxième élément dans une paire
```

> Quand nous déclarons une variable d'état avec `useState`, cela renvoie une paire–un tableau avec deux éléments. Le premier item est la valeur actuelle, et la deuxième est une fonction qui permet de la modifier. Utiliser `[0]` et `[1]` pour y accéder est un peu déconcertant puisqu'ils ont un sens spécifique. C'est pourquoi nous préférons plutôt utiliser la déstructuration positionnelle.

Reviewer 2: ```suggestion
Quand nous déclarons une variable d'état avec `useState`, cela renvoie une paire (un tableau avec deux éléments). La première valeur de la pair est la valeur actuelle, et la deuxième est une fonction qui permet de la modifier. Utiliser `[0]` et `[1]` pour y accéder est un peu déconcertant puisqu'ils ont un sens spécifique. C'est pourquoi nous préférons plutôt utiliser la déstructuration positionnelle.
```

> Quand nous déclarons une variable d'état avec `useState`, cela renvoie une paire (un tableau avec deux éléments). La première valeur de la paire est la valeur actuelle, et la deuxième est une fonction qui permet de la modifier. Utiliser `[0]` et `[1]` pour y accéder est un peu déconcertant puisqu'ils ont un sens spécifique. C'est pourquoi nous préférons plutôt utiliser la déstructuration positionnelle.

Reviewer: ```suggestion
Quand nous déclarons une variable d'état avec `useState`, ça renvoie une paire (un tableau avec deux éléments). Le premier élément est la valeur actuelle, et le deuxième est une fonction qui permet de la modifier. Utiliser `[0]` et `[1]` pour y accéder est un peu déconcertant puisqu'ils ont un sens spécifique. C'est pourquoi nous préférons plutôt utiliser la déstructuration positionnelle.
```

Tipitop™ 👏 🙇

> >Remarque :

Reviewer: ```suggestion
>Remarque
```

>
>Vous vous demandez peut-être comment React sait à quel composant `useState` fait référence étant donné que nous ne lui passons plus rien de similaire à `this`. Nous répondrons à [cette question](/docs/hooks-faq.html#how-does-react-associate-hook-calls-with-components) ainsi qu'à plein d'autres dans la section FAQ.

> ### Astuce : Utiliser plusieurs variables d'état {#tip-using-multiple-state-variables}

Reviewer: ```suggestion
### Astuce : utiliser plusieurs variables d'état {#tip-using-multiple-state-variables}
```

> Déclarer des variables d'état comme une paire de `[quelquechose, setQuelquechose]` est également pratique parce que cela nous permet de donner des noms *différents* à des variables d'état différentes si nous voulons en utiliser plus d'une :

Reviewer: ```suggestion
Déclarer des variables d'état comme une paire de `[quelquechose, setQuelquechose]` est également pratique parce que ça nous permet de donner des noms *différents* à des variables d'état différentes si nous voulons en utiliser plus d'une :
```

```js
function ExampleWithManyStates() {
  // Déclarer plusieurs variables d'état !
  const [age, setAge] = useState(42);
  const [fruit, setFruit] = useState('banane');
  const [todos, setTodos] = useState([{ text: 'Apprendre les Hooks' }]);
```

> Le composant ci-dessus, nous avons `age`, `fruit`, et `todos` comme variables locales, et nous pouvons les modifier indépendamment les unes des autres :

Reviewer 2: ```suggestion
Dans le composant ci-dessus, nous avons `age`, `fruit`, et `todos` comme variables locales, et nous pouvons les modifier indépendamment les unes des autres :
```

```js
  function handleOrangeClick() {
    // Similaire à this.setState({ fruit: 'orange' })
    setFruit('orange');
  }
```

> Utiliser plusieurs variables d'état **n'est pas obligatoire**. Les variables d'état peuvent tout à fait contenir des objets et des tableaux, vous pouvez donc toujours regrouper des données ensemble. Cependant, lorsque l'on modifie une variable d'état sa valeur est *remplacée* et non mergée, ce qui n'est pas le cas de `this.setState` dans les classes.

Reviewer: ```suggestion
Utiliser plusieurs variables d'état **n'est pas obligatoire**. Les variables d'état peuvent tout à fait contenir des objets et des tableaux, vous pouvez donc toujours regrouper des données ensemble. Cependant, lorsque l'on modifie une variable d'état sa valeur est *remplacée* et non fusionnée, contrairement à `this.setState` dans les classes.
```

> Découvrez plus de raisons de préférer séparer vos variables d'état [dans la FAQ](/docs/hooks-faq.html#should-i-use-one-or-many-state-variables).

Reviewer: ```suggestion
Découvrez les raisons de préférer séparer vos variables d'état [dans la FAQ](/docs/hooks-faq.html#should-i-use-one-or-many-state-variables).
```

## Prochaines étapes {#next-steps}

> Sur cette page, vous avez appris sur un des Hooks fournis par React, appelé `useState`. Nous nous y référons parfois par le terme « Hook State ». Il nous permet d'ajouter un état local dans une fonction composant React–ce qui est une première !

Reviewer 2: ```suggestion
Dans cet article, vous avez appris à utiliser un des Hooks fournis par React, appelé `useState`. Nous nous y référons parfois par le terme « Hook State ». Il nous permet d'ajouter un état local dans une fonction composant React–ce qui est une première !
```

> Dans cet article, vous avez appris à utiliser un des Hooks fournis par React, appelé `useState`. Nous nous y référons parfois par le terme « Hook State ». Il nous permet d'ajouter un état local dans une fonction composant React–ce qui est une première !

Reviewer: ```suggestion
Dans cette section, vous avez appris à utiliser un des Hooks fournis par React, appelé `useState`. Nous y ferons parfois référence sous le terme « Hook d’effet ». Il nous permet d'ajouter un état local à des fonctions composants React–ce qui est une première !
```

Author: Bien tenté, mais je n'ai pas perdu le fil : je pense que tu voulais écrire Hook d'état :) je corrige ça avec tes autres modifs !

Author: Autrement, je ne savais pas où l'écrire alors je le fais ici : un grand merci pour ces retours très positifs et très instructifs :) ça fait plaisir de lire ça après une toute première traduction!

Reviewer: Oops en effet, j'ai corrigé la suggestion

Reviewer: Ah ben suis-je bête tu as corrigé toute seule 😐

> Nous avons également un peu appris sur ce que sont les Hooks. Les Hooks sont des fonctions qui nous permettent de nous « brancher » à des fonctionnalités React depuis des fonctions composants. Leur noms commencent toujours par `use`, et il y a encore beaucoup de Hooks que nous n'avons pas encore vus !

Reviewer 2: ```suggestion
Nous en savons également un peu plus sur ce que sont les Hooks. Les Hooks sont des fonctions qui nous permettent de nous « brancher » à des fonctionnalités React depuis des fonctions composants. Leur noms commencent toujours par `use`, et il y a encore beaucoup de Hooks que nous n'avons pas encore vus !
```

> Nous en savons également un peu plus sur ce que sont les Hooks. Les Hooks sont des fonctions qui nous permettent de nous « brancher » à des fonctionnalités React depuis des fonctions composants. Leur noms commencent toujours par `use`, et il y a encore beaucoup de Hooks que nous n'avons pas encore vus !

Reviewer: ```suggestion
Nous en savons également appris un peu plus sur la nature des Hooks. Les Hooks sont des fonctions qui nous permettent de « nous brancher » sur des fonctionnalités React depuis des fonctions composants. Leur noms commencent toujours par `use`, et il y a encore beaucoup de Hooks que nous n'avons pas encore vus !
```

> **Continuons maintenant en [découvrant le prochain Hook : `useEffect`.](/docs/hooks-effect.html)** Il vous permet de déclencher des effets de bord dans les composants, et est similaire aux méthodes de cycles de vie dans les classes.

Reviewer: ```suggestion
**Continuons maintenant en [découvrant le prochain Hook : `useEffect`.](/docs/hooks-effect.html)** Il vous permet de déclencher des effets de bord dans les composants, et est similaire aux méthodes de cycle de vie dans les classes.
```
